src/CodeGen/ODataT4CodeGenFilesManager.ttinclude

```diff
@@ -0,0 +1,195 @@
+/// multiple source files.
+/// </summary>
+/// <param name="context">The code generation context.</param>
+[SecurityCritical]
+public class FilesManager {
+    
+        private class Block {
+            public String Name;
+            public int Start, Length;
+            public bool IsContainer;
+        }
+
+    private Block currentBlock;
+    private List<Block> files = new List<Block>();
+    private Block footer = new Block();
+    private Block header = new Block();
+    private ITextTemplatingEngineHost host;   
+    protected List<String> generatedFileNames = new List<String>();
+    public StringBuilder template{get; set;}
+
+    [SecurityCritical]
+    public static FilesManager Create(ITextTemplatingEngineHost host, StringBuilder template) {
+        return (host is IServiceProvider) ? new VSManager(host, template) : new FilesManager(host, template);
+    }
+
+    public void StartNewFile(String name, bool isContainer) {
+        if (name == null)
+            throw new ArgumentNullException("name");
+        CurrentBlock = new Block { Name = name, IsContainer =  isContainer};
+    }
+
+    public void StartFooter() {
+        CurrentBlock = footer;
+    }
+
+    public void StartHeader() {
+        CurrentBlock = header;
+    }
+
+    public void EndBlock() {
+        if (CurrentBlock == null)
+            return;
+        CurrentBlock.Length = template.Length - CurrentBlock.Start;
+        if (CurrentBlock != header && CurrentBlock != footer)
+            files.Add(CurrentBlock);
+        currentBlock = null;
+    }
+
+    public virtual void Process(bool split) {
+        if (split) {
+            EndBlock();
+            String headerText = template.ToString(header.Start, header.Length);
+            String footerText = template.ToString(footer.Start, footer.Length);
+            String outputPath = Path.GetDirectoryName(host.TemplateFile);
+            files.Reverse();
+            foreach(Block block in files) {
+                if(block.IsContainer) continue;
+                String fileName = Path.Combine(outputPath, block.Name);
+                String content = headerText + template.ToString(block.Start, block.Length) + footerText;
+                generatedFileNames.Add(fileName);
+                CreateFile(fileName, content);
+                template.Remove(block.Start, block.Length);
+            }
+        }
+    }
+
+    protected virtual void CreateFile(String fileName, String content) {
+        if (IsFileContentDifferent(fileName, content))
+            File.WriteAllText(fileName, content);
+    }
+
+    public virtual String GetCustomToolNamespace(String fileName) {
+        return null;
+    }
+
+    public virtual String DefaultProjectNamespace {
+        get { return null; }
+    }
+
+    protected bool IsFileContentDifferent(String fileName, String newContent) {
+        return !(File.Exists(fileName) && File.ReadAllText(fileName) == newContent);

```

Comment:
```suggestion
        return File.Exists(fileName) && File.ReadAllText(fileName) != newContent;
```
Easier to read

Reply:
Your suggested change will return `false` if the file does not exist. That is not what we expect.